Python/specialize.c

```diff
@@ -355,6 +355,28 @@ static int function_kind(PyCodeObject *code);
 static bool function_check_args(PyObject *o, int expected_argcount, int opcode);
 static uint32_t function_get_version(PyObject *o, int opcode);
 
+#ifdef Py_GIL_DISABLED
+static void
+maybe_enable_deferred_ref_count(PyObject *dict, PyObject *name)
+{
+    PyObject *op;
+    if (PyDict_GetItemRef(dict, name, &op) != 1) {
+        return;
+    }
+    if (_Py_IsOwnedByCurrentThread(op) ||
+            !PyType_IS_GC(Py_TYPE(op)) ||
+            _PyObject_HasDeferredRefcount(op)) {
+        Py_DECREF(op);
+        return;
+    }
```

Comment:
The `!PyType_IS_GC` and `_PyObject_HasDeferredRefcount` checks aren't really necessary because `PyUnstable_Object_EnableDeferredRefcount` handles those cases internally.